hoomd/md/many_body.py

```diff
@@ -0,0 +1,244 @@
+            \end{eqnarray*}
+
+    where the two body potential is rewritten as:
+
+    .. math::
+        :nowrap:
+
+            \begin{eqnarray*}
+            \hat{v}^{ \left( 2b \right)}_{ij}\left(\vec{r}_{ij}\right) =
+            \begin{cases}
+            & 1 \qquad \qquad \; \; \qquad r\le r_{min}\\
+            & - \dfrac{v_{ij}\left(\vec{r}_{ij}\right)}{\epsilon} \qquad r > r_{min}~.\\
+            \end{cases}
+            \end{eqnarray*}
+
+    .. attention::
+
+        The RevCross potential models an asymmetric interaction between two different chemical moieties that can form a reversible bond.
+        This requires the definition of (at least) two different types of particles.
+        A reversible bond is only possible between two different species, otherwise :math:`v^{\left( 3b \right)}_{ijk}`, would prevent any bond.
+        In our example we then set the interactions for types A and B with ``potRevC.params[(['A','B'],['A','B'])] = dict(sigma=0.0,n=0,epsilon=0,lambda3=0)``
+        and the only non-zero energy only between the different types ``potRevC.params[('A','B')] = dict(sigma=1,n=100,epsilon=100,lambda3=1)``.
+        Notice that the number of the minoritary species corresponds to the maximum number of bonds.
+
+
+    This three-body term also tunes the energy required for a bond swap through the coefficient:
+    - :math:`\lambda` - *lambda3* (unitless)
+    in `S. Ciarella and W.G. Ellenbroek 2019 <https://arxiv.org/abs/1912.08569>`_ is explained that setting :math:`\lambda=1` corresponds to no energy requirement to initiate bond swap, while this
+    energy barrier scales roughly as :math:`\beta \Delta E_\text{sw} =\beta \varepsilon(\lambda-1)`.
+
+    Note:
+
+        Choosing :math:`\lambda=1` pushes the system towards clusterization because the three-body term is not enough to
+        compensate the energy of multiple bonds, so it may cause unphysical situations.
+
+
+    Example::
+
+        nl = md.nlist.Cell()
+        potBondSwap = md.many_body.RevCross(r_cut=1.3,nlist=nl)
+        potBondSwap.params[(['A','B'],['A','B'])] = dict(sigma=0,n=0,epsilon=0,lambda3=0)
+        # a bond can be made only between A-B and not A-A or B-B
+        potBondSwap.params[('A','B')] = dict(sigma=1,n=100,epsilon=10,lambda3=1)
+    """
+    _cpp_class_name = "PotentialRevCross"
+    def __init__(self, nlist, r_cut=None, r_on=0., mode='none'):
+        super().__init__(nlist, r_cut, r_on, mode);
+        params = TypeParameter('params', 'particle_types',
+                               TypeParameterDict(sigma=2.0, n=1.0, epsilon=1.0,
+                                   lambda3=1.0, len_keys=2))
+        self._add_typeparam(params)
+
+    def attach(self, sim):
+        super().attach(sim)
+        self.nlist._cpp_obj.setStorageMode(
+            _md.NeighborList.storageMode.full)
```

Comment:
We do this for all 3 body potentials. It may be worth making a `_ThreeBody` class that does this automatically.